internal/core/app_status_report.go

```diff
@@ -0,0 +1,221 @@
+package core
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/zclconf/go-cty/cty"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/go-argmapper"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/hcl/v2"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/component"
+	sdk "github.com/hashicorp/waypoint-plugin-sdk/proto/gen"
+	"github.com/hashicorp/waypoint/internal/config"
+	pb "github.com/hashicorp/waypoint/internal/server/gen"
+)
+
+// Builds a status report on the given deployment
+func (a *App) StatusReport(
+	ctx context.Context,
+	deployTarget *pb.Deployment,
+	releaseTarget *pb.Release,
+) (*pb.StatusReport, *sdk.StatusReport, error) {
+	// Query the artifact
+	var artifact *pb.PushedArtifact
+	if deployTarget != nil && deployTarget.Preload != nil &&
+		deployTarget.Preload.Artifact != nil {
```

Comment:
I think we can avoid this last `deployTarget.Preload.Artifact != nil` check and get the same result right?

Reply:
No, we need the extra check because deployTarget might not of been preloaded with the passed through protobuf message, meaning just checking on Artifact will cause a panic with Preload being nil. You'll find a similar check in app release.

Reply:
right, I'm saying we could keep `deployTarget != nil && deployTarget.Preload != nil` and leave off the last check because if it's nil, then `var artifact *pb.PushedArtifact` will just be assigned nil anyway. Does that make sense?

Reply:
Ah, I see what you're saying. I had read that comment as the opposite. Yeah makes sense 👍🏻